index.js

```diff
@@ -0,0 +1,178 @@
+const Redis = require('ioredis')
+const CLUSTER_PREFIX = 'cluster:'
+const CHANNELS_PREFIX = 'channels:'
+const CHANNELS_KEY = 'channels'
+
+module.exports = function createClient (options = {}) {
+  const {
+    redisUrl = 'redis://localhost:6379'
+  } = options
+
+  const redis = new Redis(redisUrl)
+  let closed = false
+
+  const preventReuse = fn => (...args) => {
```

Comment:
is this to avoid trying to track a connection state in the redis client itself?

Reply:
Ah, look like I didn't need to do this at all, I was just testing improperly.

Reply:
oh, ha. well i'm all for innocent questions that end up leading to code removal :grin: